### PR TITLE
Fixes footstep sounds for downed/lying mobs

### PR DESCRIPTION
**Pull request by nero1024**
## About The Pull Request
Nobody made a report but I noticed this issue: Downed players being dragged are playing footstep sounds, specifically the bare footstep sounds via their hands from what I noticed.

Fixes (hopefully) footstep sounds from being played from incapacitated mobs that are being moved (via dragging for example). Instead, lying mobs will play a crawling sound when moving.

## Why It's Good For The Game
Footstep sounds shouldn't be played when mobs are being dragged.

## Changelog
:cl: Nero1024
fix: lying or otherwise incapacitated mobs shouldn't play footstep sounds anymore while moving or being moved
soundadd: instead they'll play a crawling/dragging sound
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Does this also stop non-slipped mobs from making footsteps? Because if someone just drags you along while you're still standing, presumably you're walking with them.

**Comment by nero1024** [comment]
Footstep sounds should still play while standing and being dragged.

File: code/datums/components/footstep.dm [hunk 1]
```diff
@@ -1,102 +1,102 @@
...
+			var/mob/living/carbon/human/H = LM
+			var/feetCover = (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET))
+			
+			if(H.shoes || feetCover) //are we wearing shoes
+				playsound(T, pick(GLOB.footstep[T.footstep][1]),
+					GLOB.footstep[T.footstep][2] * v,
+					TRUE,
+					GLOB.footstep[T.footstep][3] + e)
+			
+			if(((!H.shoes && !feetCover) || !(H.mobility_flags & MOBILITY_STAND) && !(LM.throwing || LM.movement_type & (VENTCRAWLING | FLYING) || (LM.stat != CONSCIOUS) || LM.incapacitated() || LM.IsStun() || LM.IsParalyzed()))) //are we NOT wearing shoes or are we lying/crawling (using hands to move around) and not immobilized/incapacitated?
```
Comment (ShizCalev) [inline code/datums/components/footstep.dm]: i love giant if chains

Comment (nero1024) [reply code/datums/components/footstep.dm]: I wish I didn't have to rely on them, is there a magic condition to check if someone is being dragged?

Comment (ShizCalev) [reply code/datums/components/footstep.dm]: pulledby var

**Comment by ATH1909** [comment]
Maybe we could add a dragging sound?

File: code/datums/components/footstep.dm (hunk 1 above)
Comment (nero1024) [reply code/datums/components/footstep.dm]: Oh yeah, I also still need to have the footstep sounds play while being dragged, just not when they're downed and lying on the floor in crit and the like. I would like to use the mobility flags but I can't seem to get the results I want from them hence the awkward if chains.

**Comment by nero1024** [comment]
One thing I want to know is why the footstep proc isn't returning at `if(!T.footstep || LM.buckled || !CHECK_MULTIPLE_BITFIELDS(LM.mobility_flags, MOBILITY_STAND | MOBILITY_MOVE) || LM.throwing || LM.movement_type & (VENTCRAWLING | FLYING))` like it's supposed to? The human is downed and incapacitated and should pass this if statement.

**Comment by nero1024** [comment]
I think I know what's causing the sounds, It's something to do with the mob being in the crawling state. The problem is how to check for this. You would think the mobility bitflags would do the job but they're not working for me.

**Comment by mattroks101** [comment]
I just checked the `lying` var on my copy of interstation13 to fix this. Doesn't stunning, paralyzing, weakening and knocking over all make the mob lying? If they're lying should they be playing any foostep sounds when they're moved?

**Comment by nero1024** [comment]
Not really they shouldn't play footsteps when lying.

File: code/datums/components/footstep.dm (hunk 1 above)
Comment (ShizCalev) [reply code/datums/components/footstep.dm]: the whole mobility flag thing for carbons need a rework since kevinz's shit was a clusterfuck tbh.

**Comment by Jordie0608** [comment]
You've converted the file line endings from LF to CRLF, revert them please.

File: code/datums/components/footstep.dm [hunk 2]
```diff
@@ -18,7 +18,10 @@
 	var/mob/living/LM = parent
 	var/v = volume
 	var/e = e_range
-	if(!T.footstep || LM.buckled || !CHECK_MULTIPLE_BITFIELDS(LM.mobility_flags, MOBILITY_STAND | MOBILITY_MOVE) || LM.throwing || LM.movement_type & (VENTCRAWLING | FLYING))
+	if(!T.footstep || LM.buckled || LM.lying || !CHECK_MULTIPLE_BITFIELDS(LM.mobility_flags, MOBILITY_STAND | MOBILITY_MOVE) || LM.throwing || LM.movement_type & (VENTCRAWLING | FLYING))
```
Comment (optimumtact) [inline code/datums/components/footstep.dm]: this some seriously fucked checks now

**Pull request merged**